### ISSUE TITLE
PathPrefix auth rule not working when Host rule for root is allow

### ISSUE THREAD

**Issue by fcortes** [thomseddon/traefik-forward-auth#101]
I am trying to authenticate all requests that start with '/config' but allow all other requests in the domain. I'm currently trying to achieve this with the following rules

```
rule.allow-base.action=allow
rule.allow-base.rule=Host(`my.domain.com`)

rule.auth-config.action=auth
rule.auth-config.rule=Host(`my.domain.com`) && PathPrefix(`/config`)
```

The base url works fine, but when I try to go to `/config`, it redirects to `/_oauth` and it stops there. I think that this is because the allow rule is getting handled with higher priority so the request is just forwarded to the backend and the authentication never happens.

I was thinking on having the base rule be something like `Host(`my.domain.com`) && PathPrefixRegex(`((?!oauth).)`)` but this feels more like a hack than a real solution. Is there another way to achieve this?

**Comment by SuperSandro2000** [comment]
You could put that path on a different subdomain and route that trough traefik with pathprefix.

**Comment by thomseddon** [comment]
This is a good issue, I think the only solutions at the moment are "workarounds" such as your regex or using a different domain as @SuperSandro2000 mentions.

I would like to fix this, I have written a failing test here: https://github.com/thomseddon/traefik-forward-auth/commit/320e128a6ef8fdebe3aa44bc61cf3c273f19bdda and I will have a think about possible solutions.

**Comment by fcortes** [comment]
Ok, I think I found a way to fix it, but I am not completely sure if it breaks something else (at least all other tests pass).

By adding the auth callback before adding the rules, it seems that the router correctly handles _oauth actions regardless of the additional rules. 

The test won't pass though because the scheme is set to http instead of https.

**Comment by thomseddon** [comment]
Yep, that's exactly what I was thinking - the only problem is that the current setup allows you to make a rule to prevent the "/_oauth" path from doing authorisation.

So I think this would need a major version bump, but I think it probably makes more sense
